sw/device/lib/base/mmio.c

```diff
@@ -4,6 +4,117 @@
 
 #include "sw/device/lib/base/mmio.h"
 
+#include "sw/device/lib/base/memory.h"
+
+/**
+ * Computes how many bytes `addr` is ahead of the previous 32-bit word alignment
+ * boundary.
+ */
+static ptrdiff_t misalignment32_of(uintptr_t addr) {
+  return (addr & (alignof(uint32_t) - 1));
```

Comment:
For "include what you use", you'll need to include `<stdalign.h>` to use `alignof` here - it would be good to do so.

Reply:
Small nit to add to this: `return addr % alignof(uint32_t)` is slightly more readable (the compiler will emit the `andi` code you already wrote, anyway).

Reply:
Done.